test/Microsoft.NET.Sdk.Razor.Tests/MvcBuildIntegrationTestLegacy.cs

```diff
@@ -113,9 +113,9 @@ public void Build_ProducesDepsFileWithCompilationContext_ButNoReferences()
             depsFile.Should().Exist();
             var dependencyContext = ReadDependencyContext(depsFile.FullName);
 
-            // Ensure some compile references exist
-            var packageReference = dependencyContext.CompileLibraries.First(l => l.Name == "System.Runtime.CompilerServices.Unsafe");
-            packageReference.Assemblies.Should().NotBeEmpty();
+            // Ensure compile references from a PrivateAssets="all" PackageReference don't exist
+            var packageReference = dependencyContext.CompileLibraries.FirstOrDefault(l => l.Name == "System.Runtime.CompilerServices.Unsafe", defaultValue: null);
```

Comment:
This test's project file has these items:
```
<ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.App" Version="2.2.6" PrivateAssets="All" />
    <PackageReference Include="Microsoft.AspNetCore.Razor.Design" Version="2.2.0" PrivateAssets="All" />
  </ItemGroup>
```

and expects their (indirect) references to appear in the deps.json file. They don't appear in the output, so there's no reason they should be in the deps.json.

There are similar tests that don't include the PrivateAssets. Their (indirect) references appear in the deps.json and the output directory as expected, so they don't need to be changed.